### PR TITLE
fix(amf):  fix for the issue 8742

### PR DESCRIPTION
**Pull request by VinayKumar1591**
fix(amf):  fix for the issue 8742

## Summary

1. Stored PTI in smf context within amf_context
2. Removed duplicate smf context within amf_context

## Test Plan

1. Tested with real UE and Tera VM

### CODE REVIEW

File: lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp [hunk 1]
```diff
@@ -747,12 +766,14 @@ int amf_app_handle_pdu_session_accept(
   smf_msg->header.pdu_session_id = pdu_session_resp->pdu_session_id;
   smf_msg->header.message_type   = PDU_SESSION_ESTABLISHMENT_ACCEPT;
   // smf_msg->header.procedure_transaction_id = smf_ctx->smf_proc_data.pti.pti;
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp]: you can remove the commented line as it is duplicate with the line you added next.

File: lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp [hunk 2]
```diff
@@ -747,12 +766,14 @@ int amf_app_handle_pdu_session_accept(
...
-  smf_msg->header.procedure_transaction_id = 0x01;
+  smf_msg->header.procedure_transaction_id = smf_ctx->smf_proc_data.pti.pti;
   smf_msg->msg.pdu_session_estab_accept.extended_protocol_discriminator
       .extended_proto_discriminator = M5G_SESSION_MANAGEMENT_MESSAGES;
   smf_msg->msg.pdu_session_estab_accept.pdu_session_identity.pdu_session_id =
       pdu_session_resp->pdu_session_id;
-  smf_msg->msg.pdu_session_estab_accept.pti.pti = 0x01;
+  smf_msg->msg.pdu_session_estab_accept.pti.pti =
+      smf_ctx->smf_proc_data.pti.pti;
+  ;
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp]: remove this empty statement

File: lte/gateway/configs/gateway.mconfig [hunk 3]
```diff
@@ -30,7 +30,7 @@
       "amf_name": "AMF_1",
       "amfRegionId": "1",
       "amfSetId": "1",
-      "amfPointer": "0",
+      "amfPointer": "3",
```
Comment (ulaskozat) [inline lte/gateway/configs/gateway.mconfig:33]: what is the reason for this change?

Comment (VinayKumar1591) [reply lte/gateway/configs/gateway.mconfig:33]: UE is accepting registration accept after changing amfPointer to non zero value

**Review by ulaskozat — APPROVED** [review]
LGTM
